python/ray/data/_internal/progress_bar.py

```diff
@@ -119,16 +120,25 @@ def fetch_until_complete(self, refs: List[ObjectRef]) -> List[Any]:
     def set_description(self, name: str) -> None:
         if self._bar and name != self._desc:
             self._desc = name
-            self._bar.set_description(self._desc)
+            formatted_progress = format_num(self._progress)
+            formatted_total = (
+                format_num(self._bar.total) if self._bar.total is not None else "??"
```

Comment:
here and elsewhere, i don't think we need the special case for handling if `self._bar_total is None`. according to [tqdm docs](https://tqdm.github.io/docs/tqdm/#format_meter), this is already handled by not showing the denominator (only basic stats showing numerator and rows/s):

> The expected total number of iterations. If meaningless (None), only basic progress statistics are displayed (no ETA).

Reply:
Since I update the progress bar manually, I need to make sure the number or None pass to format_num handled correctly. So I add ?? here. As when None is passed to format_num, it will raise error.

Reply:
what if we change to:
```
formatted_total = (
                format_num(self._bar.total) if self._bar.total is not None else self._bar.total
```
so we only apply `format_num` if `self._bar_total` is not None.